### PR TITLE
Only apply top level domain blocking in aggressive mode

### PR DESCRIPTION
**Pull request by pilgrim-brave**
Resolves https://github.com/brave/brave-browser/issues/15095

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_shields/browser/domain_block_navigation_throttle.cc [hunk 1]
```diff
@@ -97,9 +101,22 @@ DomainBlockNavigationThrottle::WillStartRequest() {
...
+  // Only block if Brave Shields is enabled and tracker/ad blocking is set to
+  // aggressive.
+  content::WebContents* web_contents = handle->GetWebContents();
+  HostContentSettingsMap* host_content_settings_map =
+      HostContentSettingsMapFactory::GetForProfile(
+          Profile::FromBrowserContext(web_contents->GetBrowserContext()));
+  if (!brave_shields::GetBraveShieldsEnabled(host_content_settings_map,
+                                             request_url))
+    return content::NavigationThrottle::PROCEED;
+  LOG(ERROR) << brave_shields::GetAdControlType(host_content_settings_map, request_url);
```
Comment (antonok-edm) [inline components/brave_shields/browser/domain_block_navigation_throttle.cc:113]: You left a `LOG` here :smiley:

File: browser/brave_shields/domain_block_page_browsertest.cc [hunk 2]
```diff
@@ -331,9 +345,35 @@ IN_PROC_BROWSER_TEST_F(DomainBlockTest, NoThirdPartyInterstitial) {
...
+  SetAdControlType(content_settings(), ControlType::BLOCK, url);
+
+  // Navigate to a page on a.com. This should work normally.
+  NavigateTo(url);
+  ASSERT_FALSE(IsShowingInterstitial());
+
+  // Block a.com in rules but allow a.com via shields, then attempt to navigate
+  // to a page on a.com. This should not show an interstitial.
+  BlockDomainByURL(url);
+  SetAdControlType(content_settings(), ControlType::ALLOW, url);
```
Comment (antonok-edm) [inline browser/brave_shields/domain_block_page_browsertest.cc:360]: Might be helpful to have a test for "normal" ads/trackers blocking mode (i.e. `ControlType::BLOCK_THIRD_PARTY`)

Comment (pilgrim-brave) [reply browser/brave_shields/domain_block_page_browsertest.cc:362]: added in https://github.com/brave/brave-core/commit/7ea545515d48b6a3678cbd1244d568a312dc6826

File: components/brave_shields/browser/domain_block_navigation_throttle.cc (hunk 1 above)
Comment (pilgrim-brave) [reply components/brave_shields/browser/domain_block_navigation_throttle.cc]: removed in https://github.com/brave/brave-core/commit/7ea545515d48b6a3678cbd1244d568a312dc6826

File: browser/brave_shields/domain_block_page_browsertest.cc (hunk 2 above)
Comment (antonok-edm) [reply browser/brave_shields/domain_block_page_browsertest.cc:362]: hmm, I still can't find it - am I missing something? :confused:

**Review by bridiver — APPROVED** [review]

File: browser/brave_shields/domain_block_page_browsertest.cc (hunk 2 above)
Comment (pilgrim-brave) [reply browser/brave_shields/domain_block_page_browsertest.cc:362]: https://github.com/brave/brave-core/blob/master/components/brave_shields/browser/brave_shields_util.cc#L156 so I'm calling https://github.com/brave/brave-core/blob/master/components/brave_shields/browser/brave_shields_util.cc#L127 instead.

Comment (antonok-edm) [reply browser/brave_shields/domain_block_page_browsertest.cc:362]: Gotcha! Looks good then.

**Review by antonok-edm — APPROVED** [review]

**Comment by kjozwiak** [comment]
Quickly went through this and it doesn't seem like it's working. Visiting either  `1-1ads.com`, `actionsplash.com` or `pub.chez.com`, will still load the `interstitial` page which indicates that top level domain blocking is still enabled. @pilgrim-brave is aware and is taking a look 👍

**Pull request merged**